src/data/real/ereal.lean

```diff
@@ -86,6 +91,21 @@ instance has_coe_ennreal : has_coe ℝ≥0∞ ereal := ⟨ennreal.to_ereal⟩
 instance : has_zero ereal := ⟨(0 : ℝ)⟩
 instance : inhabited ereal := ⟨0⟩
 
+/-- Negation on `ereal` -/
+protected def neg : ereal → ereal
+| ⊥       := ⊤
+| ⊤       := ⊥
+| (x : ℝ) := (-x : ℝ)
+
+instance : has_neg ereal := ⟨ereal.neg⟩
+
+/-- Subtraction on `ereal`, defined by `x - y = x + (-y)`. Since addition is badly behaved at some
+points, so is subtraction. There is no standard algebraic typeclass involving subtraction that is
+registered on `ereal` because of this bad behavior. -/
+protected noncomputable def sub (x y : ereal) : ereal := x + (-y)
+
+noncomputable instance : has_sub ereal := ⟨ereal.sub⟩
```

Comment:
If you're going to start reordering the file like this, can you split the PR?

Reply:
I don't have time for the next several days unfortunately.

Reply:
I've moved `has_sub` and `has_neg` back, but there's still some inevitable shuffling. I think that's acceptable because this is a cleanup PR.